pallets/simple-nft/src/lib.rs

```diff
@@ -56,7 +72,48 @@ pub mod pallet {
     pub struct Pallet<T>(PhantomData<T>);
 
     #[pallet::hooks]
-    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_runtime_upgrade() -> frame_support::weights::Weight {
+            frame_support::debug::info!("Upgrading runtime");
+
+            let last_token = LastToken::<T>::get();
+            let mut token_id = T::TokenId::default();
+
+            // Read all the tokens into a map
+            let mut old_tokens: BTreeMap<T::TokenId, OldToken<T::AccountId, T::TokenId, T::BlockNumber, T::TokenMetadata>> = BTreeMap::new();
+            while token_id < last_token {
+                token_id = token_id + One::one();
+                let key_hash = token_id.blake2_128_concat();
+                let old_token: OldToken<T::AccountId, T::TokenId, T::BlockNumber, T::TokenMetadata> = take_storage_value(b"SimpleNFTModule", b"TokensById", &key_hash).unwrap();
+                old_tokens.insert(token_id, old_token);
+            }
+
+            // Update and replace the token in storage
+            for (token_id, old_token) in old_tokens.iter() {
+                let key_hash = token_id.blake2_128_concat();
+                let new_token = Token {
+                    id: token_id,
+                    owner: old_token.owner,
+                    creator: old_token.creator,
+                    created_at: old_token.block_number,
+                    destroyed_at: match old_token.children.as_ref() {
+                        None => { None }
+                        Some(arr) => { 
+                            if arr.len() > 0 { Some(old_tokens.get(&arr[0]).unwrap().block_number) }
+                            else { Some(<frame_system::Module<T>>::block_number()) }
+                        }
+                    },
+                    metadata: old_token.metadata,
+                    parents: old_token.parents,
+                    children: old_token.children,
+                };
+
+                put_storage_value(b"SimpleNFTModule", b"TokensById", &key_hash, new_token);
+            }
+
+            0
```

Comment:
I think we should make an attempt at getting this right rather than just returning `0`. Have you seen any indication of best practice for coming up with the correct weight?